app/backend/chat_history/cosmosdb.py

```diff
@@ -0,0 +1,198 @@
+import os
+import time
+from typing import Any, Dict, Union
+
+from azure.cosmos.aio import ContainerProxy, CosmosClient
+from azure.identity.aio import AzureDeveloperCliCredential, ManagedIdentityCredential
+from quart import Blueprint, current_app, jsonify, request
+
+from config import (
+    CONFIG_CHAT_HISTORY_COSMOS_ENABLED,
+    CONFIG_COSMOS_HISTORY_CLIENT,
+    CONFIG_COSMOS_HISTORY_CONTAINER,
+    CONFIG_CREDENTIAL,
+)
+from decorators import authenticated
+from error import error_response
+
+chat_history_cosmosdb_bp = Blueprint("chat_history_cosmos", __name__, static_folder="static")
+
+
+@chat_history_cosmosdb_bp.post("/chat_history")
+@authenticated
+async def post_chat_history(auth_claims: Dict[str, Any]):
+    if not current_app.config[CONFIG_CHAT_HISTORY_COSMOS_ENABLED]:
+        return jsonify({"error": "Chat history not enabled"}), 405
+
+    container: ContainerProxy = current_app.config[CONFIG_COSMOS_HISTORY_CONTAINER]
+    if not container:
+        return jsonify({"error": "Chat history not enabled"}), 405
+
+    entra_oid = auth_claims.get("oid")
+    if not entra_oid:
+        return jsonify({"error": "User OID not found"}), 401
+
+    try:
+        request_json = await request.get_json()
+        id = request_json.get("id")
+        answers = request_json.get("answers")
+        title = answers[0][0][:50] + "..." if len(answers[0][0]) > 50 else answers[0][0]
+        timestamp = int(time.time() * 1000)
+
+        await container.upsert_item(
+            {"id": id, "entra_oid": entra_oid, "title": title, "answers": answers, "timestamp": timestamp}
+        )
+
+        return jsonify({}), 201
```

Comment:
wonder whether the status code here should be based on the response from `container.upsert_item` - if the item already already exists, then it will be 200 instead of 201

Reply:
That's a nice idea! I can't figure out from the SDK return types what in the response would indicate that however, as all the function signatures just say that they return a dict: https://learn.microsoft.com/en-us/python/api/azure-cosmos/azure.cosmos.container.containerproxy?view=azure-python#azure-cosmos-container-containerproxy-upsert-item
And this example doesn't do anything with the dict: https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/cosmos/azure-cosmos/samples/document_management.py#L149-L156
So is there some listing of what key in the dict would indicate it already existing?

Reply:
I've taken a look at the dicts for a first-time ID versus a second-time ID, and I can't find any key that'd indicate whether the item already existed:
https://www.diffchecker.com/Fs0ECaUa/
So I don't think we should change the status code currently, as I'm not sure what condition to use, but happy to change in the future if you have suggestions.